.github/workflows/deploy.yml

```diff
@@ -1,10 +1,14 @@
 name: Deploy
 
 on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
   push:
     branches:
       - master
-
+  pull_request:
+    branches:
+      - "**"
```

Comment:
Can be simplified to `on: push`

Reply:
Yup, just know for sure quite a lot of pushes will fail the linting and tests (at least on my part)

Reply:
It allows feature branch results to be pushed up to sonarcloud (when that is fully added in) which is a good thing in my experience. It's useful to see what going on in my branch before I raise the PR.

Reply:
so basically on pull_request (mainly for dependabot) or after we're sure build linting and tests are working on local branches, we'll do the auto build. Then on push to master (merge from PR) we do the complete + deploy steps. Else on push alone we'd be running multiple failed build processes on Git resources. Hope that makes sense?

Reply:
One could make the arguement that all commits should build and pass tests anyway.
Using `on: push` brings it in line with all other repository behaviour, including Jenkins based pipelines, so is the right trigger to use.

Reply:
true, but being an argument, i'm definitely one that doesn't push code up on complete, but rather as I go along e.g. html templates first before logic is coded. So running tests on push will fail 99% of the time. I respectfully agree to disagree. being that it's done in every other repo doesn't quite make it right for this repo

Reply:
I'll clarify as I'm not sure if it ever got written up in the handbook or not. Having commits pass tests was something that was agreed within the TIS team previously, there are obviously unavoidable exceptions but they should be minimized.

Pushing commits with known failures can complicate debugging of issues, for example if I wanted to use `git bisect` and automatically use test success/failure as a guide then the known failing commit will stop the process and require manual intervention to determine whether it is a false positive or not. If every commit fails then the tool is much more time consuming to use, if not rendered useless.
Squashing commits when merging the PR isn't a solution to that problem either as we prefer unsquashed merge commits (to master) to maintain history and allow easy reversion of feature branches.

You might not feel that it is right for this repo but I can't see any reason to diverge other than affecting your personal workflow, which goes against the team's standard practice.

I'm happy for you to widen the reviewers and get more feedback from the wider team if you wish, but at this point I won't approve it as is.

Reply:
naa, its fine, build and run tests on '**' all push